### DISCUSSION TITLE
Extend search result fields

### DISCUSSION THREAD

**Discussion by favoyang** [verdaccio/verdaccio#2248] [Ideas]
**Is your feature request related to a problem?**

Verdaccio search endpoints (`/-/all` or `/v1/search`) return standard fields like name, version, author, license... I'd like to extend custom fields into the search result.

```
 export function prepareSearchPackage(data: Package, time: unknown): any {
   ...
       name: version.name,
+      displayName: (version as any).displayName,
       description: version.description,
```

It's a very customized request, I can leave this in a branch, but I wonder if there's a way to move this into a plugin.

**Describe the solution you'd like**

Ideally, I can get a hook point like the middleware plugin to have a callback into the prepareSearchPackage function.

```
export function prepareSearchPackage(data: Package, time: unknown): any {
...
    const version: Version = data.versions[latest];
    const versions: any = { [latest]: 'latest' };
    const pkg: any = {
     ...
    };
    if (some_hook_point) {
      some_hook_point(pkg, version, versions);
    }
```

**Describe alternatives you've considered**
Maintainers can also reject this request if it's not considered useful for others.

**Additional context**
n/a

**Comment by juanpicado** [comment]
Could you elaborate a bit more? which custom fields? where those fields would be? I have an idea what you mean but maybe you can give me more.

**Comment by favoyang** [comment]
Custom package solution that built on the top of NPM may add custom fields to the search endpoints to present custom information. For Unity package manager, it adds custom fields like `displayName`, `unityVersion` to every package meta file (package.json). This is the one I'm trying to solve.

I'm not sure this is generic enough for other usages. So I'm okay with rejection.

**Comment by juanpicado** [comment]
I'd vote to keep it, sounds interesting, unique consumer for npm search is, either `npm` or `pnpm` nowadays. Let's wait new search endpoint and see whether this fits there somehow.
